pkg/kcp/provider/aws/redisinstance/client/elastiCacheClient.go

```diff
@@ -101,6 +110,120 @@ func (c *client) DeleteElastiCacheSubnetGroup(ctx context.Context, name string)
 	return nil
 }
 
+func (c *client) DescribeElastiCacheParameterGroup(ctx context.Context, name string) ([]elasticacheTypes.CacheParameterGroup, error) {
+	out, err := c.elastiCacheSvc.DescribeCacheParameterGroups(ctx, &elasticache.DescribeCacheParameterGroupsInput{
+		CacheParameterGroupName: ptr.To(name),
+	})
+	if err != nil {
+		if awsmeta.IsNotFound(err) {
+			return []elasticacheTypes.CacheParameterGroup{}, nil
+		}
+
+		return nil, err
+	}
+
+	return out.CacheParameterGroups, nil
+}
+
+func (c *client) CreateElastiCacheParameterGroup(ctx context.Context, name, family string, tags []elasticacheTypes.Tag) (*elasticache.CreateCacheParameterGroupOutput, error) {
+	out, err := c.elastiCacheSvc.CreateCacheParameterGroup(ctx, &elasticache.CreateCacheParameterGroupInput{
+		CacheParameterGroupName:   ptr.To(name),
+		CacheParameterGroupFamily: ptr.To(family),
+		Tags:                      tags,
+		Description:               ptr.To(fmt.Sprintf("ParameterGroup for ElastiCache %s", name)),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func (c *client) DeleteElastiCacheParameterGroup(ctx context.Context, name string) error {
+	_, err := c.elastiCacheSvc.DeleteCacheParameterGroup(ctx, &elasticache.DeleteCacheParameterGroupInput{
+		CacheParameterGroupName: ptr.To(name),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *client) DescribeElastiCacheParameters(ctx context.Context, groupName string) ([]elasticacheTypes.Parameter, error) {
+	result := []elasticacheTypes.Parameter{}
+	var marker *string = nil
+	for {
+		out, err := c.elastiCacheSvc.DescribeCacheParameters(ctx, &elasticache.DescribeCacheParametersInput{
+			CacheParameterGroupName: ptr.To(groupName),
+			Marker:                  marker,
```

Comment:
If the marker is returned as non nil in `out` var, will it actually be assigned to `marker` value in next request? If not, we might get into endless loop.

Reply:
Line 170 would stop endless loop
```
                if out.Marker == nil {
			break
		}
```
But you are right, marker was not being reassigned, nice catch.
I will add`marker = out.Marker` at end of the loop